integration-test/tests/email-autofill.macos.spec.js

```diff
@@ -406,9 +406,30 @@ test.describe('macos', () => {
                 .applyTo(page)
 
             const perfPage = scannerPerf(page)
-            perfPage.navigate(constants.pages['perf-huge-regex'])
+            await perfPage.navigate(constants.pages['perf-huge-regex'])
 
             await perfPage.validateInitialScanPerf(80)
         })
+
+        /**
+         * See: https://app.asana.com/0/0/1207840563867787/f
+         *
+         * This covered a form with about 60 inputs, but none are deemed 'fillable'.
+         * In that case, we want to ensure we don't do any repeated work.
+         *
+         */
+        test('Large form without eligible inputs', async ({page}) => {
+            test.setTimeout(5000)
+            await createWebkitMocks().applyTo(page)
+            await createAutofillScript()
+                .replaceAll(macosContentScopeReplacements())
+                .platform('macos')
+                .applyTo(page)
+
+            const perfPage = scannerPerf(page)
+
+            await perfPage.navigate(constants.forms['www_ulisboa_pt_login.html'])
+            await perfPage.validateInitialScanPerf(80)
```

Comment:
this `80` is another magic number, but during testing it's more than enough to ensure another regression would be difficult. on my Laptop this metric was over 300ms, so having a low number like 80 should be sufficient